src/network/postgres_protocol_handler.cpp

```diff
@@ -1016,7 +1016,17 @@ ProcessResult PostgresProtocolHandler::ProcessStartupPacket(
     LOG_TRACE("Option value is %s", token.c_str());
     cmdline_options_[token] = value;
     if (token.compare("database") == 0) {
-      traffic_cop_->SetDefaultDatabaseName(value);
+      auto exists = catalog::Catalog::GetInstance()->CheckDatabaseExists(value);
+      if (exists) {
+          traffic_cop_->SetDefaultDatabaseName(value);
+      } else {
+          auto error_msg = "Database " + value + " doesn't exist";
+          auto error_code = "3D000"; // invalid_catalog_name
+          SendErrorResponse({
```

Comment:
Have you compared the responses you sent here with the Postgres's response? You can try to capture the packet using Wireshark.

Reply:
Yup, I have captured Postgres'  responses using wireshark. I can capture Peloton's and compare the two.